src/style-spec/expression/definitions/literal.js

```diff
@@ -58,6 +59,8 @@ class Literal implements Expression {
             return ["literal", this.value];
         } else if (this.value instanceof Color) {
             return ["rgba"].concat(this.value.toArray());
+        } else if (this.value instanceof CollatorInstantiation) {
+            return this.value.serialize();
```

Comment:
This is here for the same reason as the `Color` branch above, right?  I.e., because constant folding can cause an expression tree to contain `Literal` with `value: Color` or `value: CollatorInstantiation`?

It's pre-existing, but I think this is a bit of a wart, that `Literal` can hold a value for which there's no actual literal, and thus has to know (or, in this case, the value itself has to know) how to serialize an expression that it wouldn't be responsible for parsing.  @ChrisLoer would you mind adding a quick comment in each of these two branches just briefly noting that they're here to cover the constant-folding case?